velox/functions/sparksql/Map.cpp

```diff
@@ -32,50 +32,27 @@
 namespace facebook::velox::functions::sparksql {
 namespace {
 
-template <TypeKind kind>
-void setKeysResultTyped(
+void setKeysAndValuesResult(
     vector_size_t mapSize,
     std::vector<VectorPtr>& args,
     const VectorPtr& keysResult,
+    const VectorPtr& valuesResult,
     exec::EvalCtx& context,
     const SelectivityVector& rows) {
-  using T = typename KindToFlatVector<kind>::WrapperType;
-  auto flatVector = keysResult->asFlatVector<T>();
-  flatVector->resize(rows.end() * mapSize);
-
   exec::LocalDecodedVector decoded(context);
+
   for (vector_size_t i = 0; i < mapSize; i++) {
     decoded.get()->decode(*args[i * 2], rows);
-    // For efficiency traverse one arg at the time
+    SelectivityVector targetRows(keysResult->size(), false);
```

Comment:
Let's move this variable and toSourceRow outside of the loop to avoid unnecessary memory allocations.

Reply:
Make sense, thanks.